aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -25,7 +25,17 @@ struct CopyOp {
 #else
           dst_val = static_cast<dst_T>(static_cast<native::inter_copy_type_t<dst_T>>(src_val));
 #endif
-        });
+      });
+  }
+};
+
+template<typename dst_T>
+struct CopyOp<dst_T, bool> {
+  static void apply(Tensor& dst, const Tensor& src) {
+    CUDA_tensor_apply2<dst_T, bool>(
+      dst, src, [] __device__(dst_T & dst_val, const bool& src_val) {
+        dst_val = static_cast<dst_T>(static_cast<native::inter_copy_type_t<dst_T>>(src_val));
```

Comment:
Err, was this specialization necessary because `__ldg` doesn't have a bool overload?

Reply:
That might be true, but it's pretty shocking to me.  It's definitely better if we keep `__ldg` here. Maybe this means we should read it in as a `uint8_t`, cast it to `bool`, and then do the final static cast to destination?

Reply:
cc @ngimel

Reply:
Should we just implement __ldg for bool like we did for Half?

Reply:
I'd probably wait until there is another case.

Reply:
I don't think there's a guarantee that `sizeof(bool) == sizeof(uint8_t)`, so you cannot read it as `uint8_t`?